pkg/destroy/aws/aws.go

```diff
@@ -1149,11 +1149,16 @@ func deleteEC2SubnetsByVPC(client *ec2.EC2, vpc string, failFast bool, logger lo
 	}
 
 	for _, subnet := range results.Subnets {
-		err := deleteEC2Subnet(client, *subnet.SubnetId, logger.WithField("subnet", *subnet.SubnetId))
-		if err != nil {
-			err = errors.Wrapf(err, "deleting EC2 subnet %s", *subnet.SubnetId)
+		err2 := deleteEC2Subnet(client, *subnet.SubnetId, logger.WithField("subnet", *subnet.SubnetId))
+		if err2 != nil {
+			err2 = errors.Wrapf(err2, "deleting EC2 subnet %s", *subnet.SubnetId)
 			if failFast {
-				return err
+				return err2
+			}
+			if err == nil {
+				err = err2
```

Comment:
i'm not sure what this assignment wants to achieve, `err` is from `DescribeSubnets` call

Reply:
We only enter this loop when `err` from `DescribeSubnets` was nill.  This commit just recycles that variable for the first `deleteEC2Subnet` error.

Reply:
Aaand I need to `return err` at tge end ;).

Reply:
> Aaand I need to return err at [the] end ;).

Done with cdb8e0cdd->013ff7a70.

Reply:
So this captures first failure, eats it and returns that , while all following errors are only logged at debug.

This is a weird behavior.. do we have a problem if we just collect them and report all at the end?

Reply:
> This is a weird behavior.. do we have a problem if we just collect them and report all at the end?

Sure, but we're just going to log it and throw the error out later.

Reply:
What I have now is very similar to the existing `lastError` log-and-forget pattern we have [here][1], etc..  How extensively do you want me to rewrite this package?

[1]: https://github.com/openshift/installer/blob/83be6cf6e7cdd26986f75245002d17f6e9244960/pkg/destroy/aws/aws.go#L333-L336

Reply:
> What I have now is very similar to the existing lastError log-and-forget pattern we have here, etc

Yeah, I was looking at other functions. So let's keep this behavior consistent at least with existing for now.

But maybe can we use the lastError, err like others compared to this err shadowing, err2 to be more readable.

Reply:
> But maybe can we use the lastError...

Rerolled to use `lastError` with 013ff7a70 -> 58524d549.

Reply:
This eats the first error all together ;) but it's fine as that's what other places do